DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/Factory/AnonymizerSettingsFactory.cs

```diff
@@ -0,0 +1,30 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Processors.Settings
+{
+    public class AnonymizerSettingsFactory : IAnonymizerSettingsFactory
+    {
+        public T CreateAnonymizerSetting<T>(JObject settings)
+        {
+            EnsureArg.IsNotNull(settings, nameof(settings));
+
+            try
+            {
+                return settings.ToObject<T>();
+            }
+            catch (Exception ex)
+            {
+                throw new AnonymizerConfigurationException(DicomAnonymizationErrorCode.InvalidRuleSettings, "Fail to parse anonymizer setting.", ex);
```

Comment:
Is it anonymizer setting, rule setting or processor setting?
I feel you need to unify the terms even if they're equivalent.

Reply:
Only rule setting and anonymizer setting.
Rule setting is defined in DICOM config and Anonymizer setting is used in shared lib.
I modified to "Fail to parse rule setting" here.